src/server/cookies.ts

```diff
@@ -272,16 +291,14 @@ export function getChunkedCookie(
   }
 
   // Validate sequence integrity - check for missing chunks
-  const highestIndex = getChunkedCookieIndex(chunks[chunks.length - 1].name)!;
+  const highestIndex = getChunkedCookieIndex(
+    chunks[chunks.length - 1].name,
+    isLegacyCookie ?? false
```

Comment:
I believe we can avoid `?? false` here

```suggestion
    isLegacyCookie
```